### PR TITLE
Add a mission trigger and a condition to check if the flagship is disabled

### PR DESCRIPTION
**Pull request by TomGoodIdea**
**Feature:** This PR implements the feature request detailed and discussed in issue #8603

## Feature Details
 - Added a new mission trigger and an `"on disabled"` node. It fires up when player's flagship becomes disabled.
 - Added a condition returning 1 if the flagship is disabled.

## Testing Done
Both the trigger and the condition have been tested by inserting some conversations into a mission.

## Performance Impact
Probably none.

### CODE REVIEW

File: source/PlayerInfo.cpp
Comment (Hurleveur) [inline source/PlayerInfo.cpp:2946]: you need to respect the conventions:
{
  indented code
};

**Review by Hurleveur — CHANGES_REQUESTED** [review]
that's much simpler than I thought, looking good
thx for doing this!
just this small convention that needs to be respected (in general its good to look at similar code to know how we usually write it)

nice that you did both the on disabled and the condition because I see use cases for both

File: source/Mission.cpp
Comment (quyykk) [inline source/Mission.cpp:1150]:
```suggestion
	if((event.Type() & ShipEvent::DISABLE) && event.Target().get() == player.Flagship())
```

Comment (TomGoodIdea) [reply source/Mission.cpp:1150]: Isn't the bitwise AND evaluated before the logical AND?

Comment (quyykk) [reply source/Mission.cpp:1150]: Yeah but that's the common advice when using bitwise operations, because the order of evaluation for them is weird. It's basically done for clarity.

**Review by Hurleveur — APPROVED** [review]
Looking good to me!
Nice to have this, eventually we'll have this for the entire fleet being disabled too, I hope
In the meantime this is useful for duels

**Comment by TomGoodIdea** [comment]
@Hurleveur would you like `on disabled` changed to `on "flagship disabled"`? Or leave it as is and add `on "fleet disabled"` in the future?

**Comment by Hurleveur** [comment]
Seems like a good idea to me, but mb with another syntax? Idk tbf, but I think we dont usually do two words keywords.
Meaning it would more be like `on disabled flagship` and `on disabled fleet` if we want to base it on disabling or on flagship disabled if we think we can have other flagship actions
I think the first option is more sound but prob best to wait for another opinion

**Comment by Hurleveur** [comment]
As soon as this gets merged I'll work on doing the other option (fleet disabled) as this is going to be really helpful for the Unfettered campaign and all the duels in it.
Since these attributes work by one block anyway we could just name the one that concerns the fleet "on fleet-disabled" and be fine?

**Comment by Hurleveur** [comment]
I'd love to see this progess so that https://github.com/endless-sky/endless-sky/pull/8828 can actually make you fail the duel without dying
So @quyykk could you voice which syntax is rhe best of the 2?

**Pull request merged**
